orangecontrib/text/vectorization/sbert.py

```diff
@@ -0,0 +1,125 @@
+import json
+import base64
+import zlib
+import sys
+from typing import Any, List, Optional, Callable, Union
+
+import numpy as np
+
+from Orange.misc.server_embedder import ServerEmbedderCommunicator
+from Orange.util import dummy_callback
+
+# maximum document size that we still send to the server
+MAX_PACKAGE_SIZE = 3000000
+# maximum size of a chunk - when one document is longer send is as a chunk with
+# a single document
+MAX_CHUNK_SIZE = 50000
+MIN_CHUNKS = 20
+
+
+class SBERT:
+    def __init__(self) -> None:
+        self._server_communicator = _ServerCommunicator(
+            model_name='sbert',
+            max_parallel_requests=100,
+            server_url='https://apiv2.garaza.io',
+            embedder_type='text',
+        )
+
+    def __call__(
+        self, texts: List[str], callback: Callable = dummy_callback
+    ) -> List[Optional[List[float]]]:
+        """Computes embeddings for given documents.
+
+        Parameters
+        ----------
+        texts
+            A list of raw texts.
+
+        Returns
+        -------
+        An array of embeddings.
+        """
+
+        if len(texts) == 0:
+            return [None] * len(texts)
+
+        skipped = list()
+
+        encoded_texts = list()
+        sizes = list()
+        chunks = list()
+        for i, text in enumerate(texts):
+            encoded = base64.b64encode(zlib.compress(
+                text.encode('utf-8', 'replace'), level=-1)
+            ).decode('utf-8', 'replace')
+            size = sys.getsizeof(encoded)
+            if size > MAX_PACKAGE_SIZE:
+                skipped.append(i)
+                continue
+            encoded_texts.append(encoded)
+            sizes.append(size)
+
+        chunks = self._make_chunks(encoded_texts, sizes)
+
+        # temporary callback - will be changed when ServerEmbedderCommunicator
+        # change callback - return proportion instead bool
+        ticks = iter(np.linspace(0.0, 1.0, len(chunks)))
+
+        def cb(success=True):
+            if success:
+                callback(next(ticks))
+
+        result_ = self._server_communicator.embedd_data(chunks, callback=cb)
```

Comment:
`callback` is new embedders do not need a callback that we used before (before it was processed_callback but is deprecated). Passing just callback further is ok. In order to use callback please also consider setting the orange version on >=3.32 in requirements
```suggestion
        result_ = self._server_communicator.embedd_data(chunks, callback=callback)
```